### ISSUE TITLE
Bug: [no-floating-promises] allowForKnownSafeCalls no longer works for `describe` function with newer node:test typings

### ISSUE THREAD

**Issue by kayahr** [typescript-eslint/typescript-eslint#11372]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.8.2&fileType=.tsx&code=JYWwDg9gTgLgBAbzgEwKYGcDGVgCNUA0cw8AvnAGZQQhwBEAdhGgFwwYx0BQXaWO%2BABR0KECHSKCAlHAC8APkRc4xGMNwBDKBLjS5ihMrikpAbi4nzXIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oDN4OBDfMwDmtYtA4BbSshTowAbXDYcAd37QmkADSZlKnP3h9VAMQ7QA0p1VMAyv26IAwkfioMSgwazeDkbglJeUhifjIAa35hRB19PxwmfklYzyhKfB0oABNyKgAjWN1IfBRM4uRYDNiwAF1teITQ8KiYkM5ctFLkTMaDAF8%2BiFq%2BwZURiEH%2BoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
import { describe, it } from "node:test"

describe("foo", () => {
  it("bar", () => {
  });
});
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-floating-promises": [
      "warn", {
        "allowForKnownSafeCalls": [
          {
            "from": "package",
            "name": [ "it", "describe", "test", "suite" ],
            "package": "node:test"
          }
        ]
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

The floating-promise warning should be ignored for `describe` and `it` calls from `node:test`.

### Actual Result

Since `@types/node` version 24.0.9 it only works for `it` but no longer for `describe`. With version 24.0.8 it works for both calls.

### Additional Info

In `@types/node` version 24.0.8 the `it` and `describe` functions were exported directly. Since 24.0.9 the functions are exported as an alias:

```ts
namespace test {
    export { test };
    export { suite as describe, test as it };
}
// ...
export = test;
```

The `it` function only works when I also add the `test` function in the `allowForKnownSafeCalls` config. But adding `suite` doesn't fix the problem with `describe`. 

The export definitions in `@types/node` are syntactically correct, no compiler and runtime errors. So I guess this is a bug in `typescript-eslint` that these definitions are not parsed correctly?

**Comment by textbook** [comment]
I'm seeing the same problem between `@types/node@20.19.2` and `...@20.19.3`, where the same change to the way the types are exported was applied - DefinitelyTyped/DefinitelyTyped#73002

**Comment by Renegade334** [comment]
The issue is with `findParentModuleDeclaration()`:
https://github.com/typescript-eslint/typescript-eslint/blob/c81ba15ab439b6880157dd2bd41fc830f903e8b5/packages/type-utils/src/typeOrValueSpecifiers/typeDeclaredInPackageDeclarationFile.ts#L3-L16

Essentially, this logic doesn't play well with symbols contained within exported namespaces. Namespace declarations are also `ModuleDeclaration`s in the AST, so this function walks the AST upwards, reaches the containing namespace, returns _its_ name, and it's this which is then compared (unsuccessfully) against the configured package name from `allowForKnownSafeCalls`. With the changes to the `node:test` structure, all its named exports (except for the top-level `test` symbol) will fail this logic.

The fix should be as simple as checking for the `Namespace` flag on `ModuleDeclaration` nodes, and ignoring the node if found.

**Comment by Renegade334** [comment]
So, the exact case as reported is now fixed; however some doodling has thrown up a related hiccough:

```js
// eslint.config.mjs
...
"allowForKnownSafeCalls": [
  {
    "from": "package",
    "name": "describe",
    "package": "node:test"
  }
]
...

// test.ts
import { describe } from 'node:test';

describe(); // 🫤 
```

This also complains, for a completely separate issue: the specifier matcher checks if the type name matches the specifier name, but this logic fails with aliased exports (a variant of #10053). In this particular example, the specifier would need to be `"name": "suite"` (the alias target) to match successfully.

If this is a WONTFIX and remains a significant issue for your users, then there's no fundamental reason why we couldn't return to providing duplicated definitions for these rather than aliasing them. It doesn't solve the wider issue, but I imagine that `node:test` is probably the most common case for this specific rule option, and it wouldn't cause any significant impact for @types/node besides a minor hit to maintainability.

**Comment by kirkwaiblinger** [comment]
@Renegade334 Would you mind spinning off a new issue for this with a playground link/reproduction?

(From a quick look, I think this is similar to https://github.com/typescript-eslint/typescript-eslint/issues/10660, and I'm guessing we need to also check `valueMatchesSomeSpecifier`, rather than only the type, in order to ensure we detect the symbol named `describe` when specifying `describe` from `node:test`)
